File: android/app/src/main/java/com/desklink/android/domain/model/MessageType.kt
package com.desklink.android.domain.model

object MessageType {
    // Control channel (0x01-0x0F)
    const val HANDSHAKE_REQUEST: Byte = 0x01
    const val HANDSHAKE_RESPONSE: Byte = 0x02
    const val CONFIG_REQUEST: Byte = 0x03
    const val CONFIG_RESPONSE: Byte = 0x04
    const val START_STREAM: Byte = 0x05
    const val STOP_STREAM: Byte = 0x06
    const val PING: Byte = 0x07
    const val PONG: Byte = 0x08
    const val ERROR: Byte = 0x09
    const val DISCONNECT: Byte = 0x0A
    const val BITRATE_UPDATE: Byte = 0x0B
    const val CONFIG_UPDATE: Byte = 0x0C
    // LAN mutual authentication (P3): challenge-response keyed by HKDF(PIN), inside TLS.
    const val AUTH_CHALLENGE: Byte = 0x0D
    const val AUTH_RESPONSE: Byte = 0x0E
    const val AUTH_CONFIRM: Byte = 0x0F

    // Video channel (0x10-0x1F)
    const val VIDEO_FRAME: Byte = 0x10
    const val VIDEO_CONFIG: Byte = 0x11
    const val KEYFRAME_REQUEST: Byte = 0x12

    // Input channel (0x20-0x2F)
    const val TOUCH_EVENT: Byte = 0x20
    const val TOUCH_BATCH: Byte = 0x21
    const val SCROLL: Byte = 0x22
    const val POINTER_BUTTON: Byte = 0x23
}

File: macos/DeskLink/Sources/Domain/Models/MessageType.swift
import Foundation

public enum MessageType: UInt8, Sendable {
    case handshakeRequest = 0x01
    case handshakeResponse = 0x02
    case configRequest = 0x03
    case configResponse = 0x04
    case startStream = 0x05
    case stopStream = 0x06
    case ping = 0x07
    case pong = 0x08
    case error = 0x09
    case disconnect = 0x0A
    case bitrateUpdate = 0x0B
    case configUpdate = 0x0C
    case authChallenge = 0x0D
    case authResponse = 0x0E
    case authConfirm = 0x0F

    case videoFrame = 0x10
    case videoConfig = 0x11
    case keyframeRequest = 0x12

    case touchEvent = 0x20
    case touchBatch = 0x21
    case scroll = 0x22
    case pointerButton = 0x23
}
